### PR TITLE
Various improvements to "Review and Deploy" modal

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

- Reorganise "Review and Deploy" modal into 2 sections to indicate which can be changed instantly and not
- General layout tidy up of the modal for desktop and mobile (using `Row`/`Col` elements rather than just `Space` elements)
- Adds sentence to 'Project details' drawer 

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/152669564-b77afdfb-6361-4a17-896a-b281df5e5859.mp4

When no duration set:

https://user-images.githubusercontent.com/96150256/152669556-ec076379-3ad7-41ac-91d1-be8def5c3900.mp4

Mobile:

https://user-images.githubusercontent.com/96150256/152669590-a2a8f4d0-0376-4aff-8699-f455c73d4d9b.mp4

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

File: src/components/Create/index.tsx [hunk 1]
```diff
@@ -726,7 +726,7 @@ export default function Create() {
           }
           onOk={deployProject}
           confirmLoading={loadingCreate}
-          width={800}
+          width={600}
```
Comment (peripheralist) [inline src/components/Create/index.tsx:729]: what's the reason behind this change? the modal used to be narrower (i think 600) but was widened mainly because when things like website + pay text are set it, can run long and look awkward

i'd be happy to work some reformatting into this PR though if you have some ideas

**Comment by peripheralist** [comment]
love this change @johnnyd-eth! awesome way to clarify the "reconfig rules" for different types of project info without adding any new components or big restructuring.

i'm also down to use the row/col structure—looks nice! though i'm wondering how text overflow is handled. have you tested this with website, twitter, pay disclosure set? pay disclosure could be quite long depending on how projects are using it and should probably have its own line, i've seen some long website urls as well

File: src/components/Create/index.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/Create/index.tsx:729]: I initially dropped this down because it seemed like there was a lot of unnecessary whitespace when some fields were empty. But now that they're spread out more horizontally I do think 800px actually looks better.

**Comment by johnnyd-eth** [comment]
@peripheralist thanks for this feedback. Definitely wasn't considering long URL's and 'Payment disclosure'. I've reorganized the fields (see below): moved the disclosure to its own line, and gave 'Discord' and 'Website' a bit more space. I think in the case where these fields are short or empty it still looks decent. 

Long fields:

Screen Shot 2022-02-08 at 9 01 13 am

Short fields:

Screen Shot 2022-02-08 at 9 01 42 am

**Comment by peripheralist** [comment]
nice @johnnyd-eth ! much better. still a little concerned about scrunching on the name handle, but overall no big deal. generally i think i prefer too much whitespace (with empty fields) over overcrowding (with full fields), but it'd also be nice to keep this form from getting too vertically long. this might be the best in-between solution

will play around with it when you push your changes!

**Comment by johnnyd-eth** [comment]
Also, moved to full-width fields on mobile:

https://user-images.githubusercontent.com/96150256/152888884-89d0059b-eca6-4de5-94c1-1ff0b72f4178.mp4

**Comment by johnnyd-eth** [comment]
@peripheralist Good call, defs agree we don't want it looking crowded. Gave a bit more space to the name and pushed my latest. LMK what you think 👀

**Review by peripheralist — APPROVED** [review]
looks awesome @johnnyd-eth, nice work 👏
